From c1f847a8c02656cffabd63ffb92d044e41aa37ae Mon Sep 17 00:00:00 2001
From: Alex <alexander.paige@intel.com>
Date: Thu, 21 Mar 2019 03:47:20 -0700
Subject: [PATCH] Fixing GNUInstallDirs and adding missiing version update.

Needed to make GNUInstallDirs at a more global scope and add the updated version cmake changes to the igc-opencl.pc.in file.

Change-Id: Iff4202b15a41e606f062929d5401e38b94176135
---
 IGC/AdaptorOCL/igc-opencl.pc.in    |  2 +-
 IGC/CMakeLists.txt                 | 18 +++++++++++-------
 visa/CMakeLists.txt                |  2 +-
 visa/iga/IGAExe/CMakeLists.txt     |  2 +-
 visa/iga/IGALibrary/CMakeLists.txt |  5 +----
 5 files changed, 15 insertions(+), 14 deletions(-)

diff --git a/IGC/AdaptorOCL/igc-opencl.pc.in b/IGC/AdaptorOCL/igc-opencl.pc.in
index d211c2f..d88165b 100644
--- a/IGC/AdaptorOCL/igc-opencl.pc.in
+++ b/IGC/AdaptorOCL/igc-opencl.pc.in
@@ -5,6 +5,6 @@ libdir=${exec_prefix}/lib
 
 Name: igc-opencl
 Description: Intel(R) Graphics Compiler for OpenCL(TM)
-Version: @MAJOR_VERSION@.@MINOR_VERSION@.@PATCH_VERSION@
+Version: @IGC_API_MAJOR_VERSION@.@IGC_API_MINOR_VERSION@.@IGC_API_PATCH_VERSION@
 Cflags: -I${includedir} -I${includedir}/cif -I${includedir}/ocl_igc_shared/executable_format -I${includedir}/ocl_igc_shared/device_enqueue
 Libs:
diff --git a/IGC/CMakeLists.txt b/IGC/CMakeLists.txt
index fd6721d..3a0657c 100644
--- a/IGC/CMakeLists.txt
+++ b/IGC/CMakeLists.txt
@@ -47,6 +47,10 @@ endif()
 
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
+if(UNIX)
+  include(GNUInstallDirs)
+endif()
+
 if(POLICY CMP0058)
     cmake_policy(SET CMP0058 NEW)
 endif()
@@ -3876,13 +3880,13 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/AdaptorOCL/igc.opencl.h.in ${IGC_BUIL
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/AdaptorOCL/igc-opencl.pc.in ${IGC_BUILD__IGC_BIN_DIR}/AdaptorOCL/igc-opencl.pc @ONLY)
 
 if(UNIX)
-  install(TARGETS ${IGC_BUILD__PROJ__igc_dll} DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/lib COMPONENT igc-core)
-  install(TARGETS ${IGC_BUILD__PROJ__fcl_dll} DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/lib COMPONENT igc-opencl)
-  install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/AdaptorOCL/ocl_igc_shared DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/include/igc COMPONENT igc-opencl-devel)
-  install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/AdaptorOCL/ocl_igc_interface DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/include/igc COMPONENT igc-opencl-devel)
-  install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/AdaptorOCL/cif DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/include/igc COMPONENT igc-opencl-devel)
-  install(FILES ${IGC_BUILD__IGC_BIN_DIR}/AdaptorOCL/igc-opencl.pc DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/lib/pkgconfig COMPONENT igc-opencl-devel)
-  install(FILES ${IGC_BUILD__IGC_BIN_DIR}/AdaptorOCL/igc.opencl.h DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/include/igc COMPONENT igc-opencl-devel)
+  install(TARGETS ${IGC_BUILD__PROJ__igc_dll} LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR} COMPONENT igc-core)
+  install(TARGETS ${IGC_BUILD__PROJ__fcl_dll} LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR} COMPONENT igc-opencl)
+  install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/AdaptorOCL/ocl_igc_shared DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/igc COMPONENT igc-opencl-devel)
+  install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/AdaptorOCL/ocl_igc_interface DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/igc COMPONENT igc-opencl-devel)
+  install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/AdaptorOCL/cif DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/igc COMPONENT igc-opencl-devel)
+  install(FILES ${IGC_BUILD__IGC_BIN_DIR}/AdaptorOCL/igc-opencl.pc DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}/pkgconfig COMPONENT igc-opencl-devel)
+  install(FILES ${IGC_BUILD__IGC_BIN_DIR}/AdaptorOCL/igc.opencl.h DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/igc COMPONENT igc-opencl-devel)
 elseif (WIN32)
   install(TARGETS ${IGC_BUILD__PROJ__igc_dll}
           RUNTIME DESTINATION Release/lh64          CONFIGURATIONS Release)
diff --git a/visa/CMakeLists.txt b/visa/CMakeLists.txt
index c230f8d..8013a14 100644
--- a/visa/CMakeLists.txt
+++ b/visa/CMakeLists.txt
@@ -361,7 +361,7 @@ if (UNIX OR WIN32)
   endif()
   
   if(UNIX)
-    install(TARGETS GenX_IR_Exe DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/bin COMPONENT igc-media)
+    install(TARGETS GenX_IR_Exe RUNTIME DESTINATION ${CMAKE_INSTALL_FULL_BINDIR} COMPONENT igc-media)
   endif(UNIX)
 
   # Copy executable
diff --git a/visa/iga/IGAExe/CMakeLists.txt b/visa/iga/IGAExe/CMakeLists.txt
index 99cb3cb..1fb9a7c 100644
--- a/visa/iga/IGAExe/CMakeLists.txt
+++ b/visa/iga/IGAExe/CMakeLists.txt
@@ -106,7 +106,7 @@ if(IGC_BUILD)
     endif(CMAKE_SIZEOF_VOID_P EQUAL 8)
 
   elseif (UNIX)
-    install(TARGETS IGA_EXE DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/bin COMPONENT igc-media)
+    install(TARGETS IGA_EXE RUNTIME DESTINATION ${CMAKE_INSTALL_FULL_BINDIR} COMPONENT igc-media)
   endif (MSVC)
 else()
     INSTALL(TARGETS IGA_EXE
diff --git a/visa/iga/IGALibrary/CMakeLists.txt b/visa/iga/IGALibrary/CMakeLists.txt
index f4e9d34..7c9cb83 100644
--- a/visa/iga/IGALibrary/CMakeLists.txt
+++ b/visa/iga/IGALibrary/CMakeLists.txt
@@ -240,7 +240,6 @@ endif()
 # DISABLE_ENCODER_EXCEPTIONS: Disabling exceptions for encoder only project.
 # So it can be used in vISA BE, which is used in IGC. (IGC doesn't permit C++
 # exceptions to be used
-#
 # Alex: seems like this is the way to do it so it works in various
 # linux, android builds.
 target_compile_definitions(IGA_ENC_LIB PUBLIC DISABLE_ENCODER_EXCEPTIONS)
@@ -331,9 +330,7 @@ endif(ANDROID AND MEDIA_IGA)
 
   if(UNIX)
     INSTALL(TARGETS IGA_DLL
-        RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/.   COMPONENT igc-core
-        LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib COMPONENT igc-core
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/.   COMPONENT igc-core
+        LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR} COMPONENT igc-core
     )
   elseif(WIN32)
     install(TARGETS IGA_DLL RUNTIME
-- 
2.17.1

